app/services/ExactTargetService.scala

```diff
@@ -52,6 +52,12 @@ trait ExactTargetService extends LazyLogging {
     )
     val subscriptionDetails = getPlanDescription(validPromotion, subscriptionData.personalData.currency, subscriptionData.productRatePlanId)
     val promotionDescription = validPromotion.filterNot(_.promotion.promotionType == Tracking).map(_.promotion.description)
+    val redemptionInstructions = for {
```

Comment:
could you not do `validPromotion.promotion.asIncentive.flatMap(_.redemptionInstructions)`